pandas/core/generic.py

```diff
@@ -651,32 +650,24 @@ def _set_axis(self, axis, labels):
         self._data.set_axis(axis, labels)
         self._clear_item_cache()
 
-    def transpose(self, *args, **kwargs):
+    def transpose(self, copy, valid_axes=[1, 0]):
```

Comment:
Can you make the default `valid_axes=None` and then handle as appropriate within the method?  Mutable default arguments can lead to problems so probably best to avoid just to be safe.

Reply:
Great callout! Do you think better to define here or even just move this code to the DataFrame class?

Reply:
Not super familiar with this, so don't want my opinion to carry too much weight, but I think it'd be cleaner to push this back to the `DataFrame` class and remove it from here entirely.  If we're removing `Panel.transpose` then I think `DataFrame` is the only thing that uses this method, so seems more direct that way?  Looks like `Series`/`Index` inherit from `IndexOpsMixin` and sparse things have their own implementation, so should be safe to remove unless I'm forgetting something?

Reply:
We probably could move the "core" implementation into `DataFrame`, but I would think we should keep a skeleton implementation on `transpose` in `NDFrame` (even simple as `raise NotImplemented`, as `transpose` is arguably something that should be part of the implementation of anything that inherits from this base class).